### PR TITLE
JMPI-494 error handling

### PR DESCRIPTION
**Pull request by MatthewErispe**

### CODE REVIEW

File: JeMPI_Apps/JeMPI_API/pom.xml [hunk 1]
```diff
@@ -194,6 +194,11 @@
             <artifactId>akka-actor-testkit-typed_${scala.tools.version}</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/pom.xml:197]: There's another LibMPI dependency at line 27, I don't get the difference, is it a duplicate ?

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java [hunk 2]
```diff
@@ -221,18 +223,32 @@ private Behavior<Event> findMatchesForReviewHandler(final FindMatchesForReviewRe
...
+
+      try {
+         libMPI.startTransaction();
+         count = libMPI.countGoldenRecords();
+         libMPI.closeTransaction();
+      } catch (Exception exception) {
+         LOGGER.error("libMPI.countGoldenRecords failed with error message: {}", exception.getMessage());
+      }
+
       request.replyTo.tell(new GetGoldenRecordCountResponse(count));
```
[the remark this answers is not in the document]
Comment (MatthewErispe) [reply JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java:236]: Does the user need to know why the count is 0? This handles the case where no records exist

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java [hunk 3]
```diff
@@ -257,58 +273,144 @@ private Behavior<Event> getGoldenIdsHandler(final GetGoldenIdsRequest request) {
...
+      try {
+         libMPI.startTransaction();
+         patientRecord = libMPI.findPatientRecord(request.patientId);
+         goldenRecords = libMPI.getCandidates(patientRecord.demographicData(), true);
+         libMPI.closeTransaction();
+      } catch (Exception exception) {
+         LOGGER.error("findCandidatesHandler failed to find patientId: {}", request.patientId);
+         request.replyTo.tell(new FindCandidatesResponse(Either.left(new MpiServiceError.PatientIdDoesNotExistError(
+               "Patient not found",
+               request.patientId))));
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java:390]: I think this will be executed twice (see line 394)

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java (hunk 3 above)
Comment (MatthewErispe) [reply JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/BackEnd.java:390]: I think I forgot to remove this when refactoring. The only way to really know is to use a nested try catch. One for both libmpi calls then I can add the tell error to each catch
Are you happy with this approach?

**Review by marrouchi — APPROVED** [review]

**Pull request merged**